src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java

```diff
@@ -154,12 +157,23 @@ public EventTypeController(final EventTypeRepository eventTypeRepository, final
         }
     }
 
-    private void validatePartitioningStrategy(final EventType eventType) throws NoSuchPartitioningStrategyException {
+    private void validatePartitioningStrategy(final EventType eventType) throws NoSuchPartitioningStrategyException,
+            InvalidEventTypeException {
         final PartitionResolutionStrategy partitioningStrategy = eventType.getPartitionResolutionStrategy();
-        if (partitioningStrategy != null && !partitionResolver.strategyExists(partitioningStrategy.getName())) {
+        if (!partitionResolver.strategyExists(partitioningStrategy.getName())) {
             throw new NoSuchPartitioningStrategyException("partitioning strategy does not exist: " +
                     partitioningStrategy.getName());
         }
+        else if (HASH_STRATEGY.equals(partitioningStrategy.getName()) &&
+                eventType.getPartitioningKeyFields().isEmpty()) {
```

Comment:
Maybe we also check that the specified `PartitioningKeyFields`are valid and required fields.